insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,100 @@
+"""
+softnet_stat - /proc/net/softnet_stat
+=====================================
+
+This parser parses the stats from network devices. These stats includes events
+per cpu(in row), number of packets processed i.e packet_process (first column),
+number of packet drops packet_drops (second column), time squeeze i.e net_rx_action
+performed time_squeeze(third column),cpu collision i.e collision occur while obtaining
+device lock while transmitting cpu_collision packets (Ninth column), received_rps
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = shared[SoftNetStats]
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.get_cpu_instaces()
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        {'cpu2': '00000000', 'cpu3': '00000001', 'cpu0': '00000000', 'cpu1': '00000000'}
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def parse_content(self, contents):
+        cpu = 0
+        data = {}
+        for line in contents:
+            stats = line.split(' ')
+
+            data['cpu' + str(cpu)] = {'packet_process': stats[0],
+                                    'packet_drops': stats[1],
+                                    'time_squeeze': stats[2],
+                                    'cpu_collision': stats[8],
+                                    'received_rps': stats[9],
+                                    'flow_limit_count': stats[10]}
+            cpu = cpu + 1
+        self.data = data
+        self.data['cpu_instances'] = cpu
```

Comment:
I'm not sure what the purpose of this piece of information.  You can tell the number of CPUs by len(self.data).

If you do want to keep it, I'd also recommend removing that from `self.data` and instead make `cpu_instances` a property - e.g. `self.cpu_instances = cpu`.  Then you don't need a getter method.